lib/web_ui/lib/src/engine/canvaskit/font_fallbacks.dart

```diff
@@ -0,0 +1,690 @@
+      String googleFontCss = await html.window
+          .fetch(font.googleFontsCssUrl)
+          .then((dynamic response) =>
+              response.text().then<String>((dynamic x) => x as String));
+      final _ResolvedNotoFont resolvedFont =
+          _makeResolvedNotoFontFromCss(googleFontCss, font.name);
+      _registerResolvedFont(font, resolvedFont);
+    }
+  }
+
+  Set<_ResolvedNotoSubset> resolvedFonts = <_ResolvedNotoSubset>{};
+  for (int codeunit in codeunits) {
+    resolvedFonts.addAll(_lookupResolvedFontsForCodeunit(codeunit));
+  }
+
+  for (_ResolvedNotoSubset resolvedFont in resolvedFonts) {
+    skiaFontCollection.registerFallbackFont(
+        resolvedFont.url, resolvedFont.name);
+  }
+  await skiaFontCollection.ensureFontsLoaded();
+  sendFontChangeMessage();
+}
+
+_ResolvedNotoFont _makeResolvedNotoFontFromCss(String css, String name) {
+  List<_ResolvedNotoSubset> subsets = <_ResolvedNotoSubset>[];
+  bool resolvingFontFace = false;
+  String? fontFaceUrl;
+  List<_UnicodeRange>? fontFaceUnicodeRanges;
+  for (final String line in LineSplitter.split(css)) {
+    // Search for the beginning of a @font-face.
+    if (!resolvingFontFace) {
+      if (line == '@font-face {') {
+        resolvingFontFace = true;
+      } else {
+        continue;
+      }
+    } else {
+      // We are resolving a @font-face, read out the url and ranges.
+      if (line.startsWith('  src:')) {
+        int urlStart = line.indexOf('url(');
+        if (urlStart == -1) {
+          throw new Exception('Unable to resolve Noto font URL: $line');
+        }
+        int urlEnd = line.indexOf(')');
+        fontFaceUrl = line.substring(urlStart + 4, urlEnd);
+      } else if (line.startsWith('  unicode-range:')) {
+        fontFaceUnicodeRanges = <_UnicodeRange>[];
+        String rangeString = line.substring(17, line.length - 1);
+        List<String> rawRanges = rangeString.split(', ');
+        for (final String rawRange in rawRanges) {
+          List<String> startEnd = rawRange.split('-');
+          if (startEnd.length == 1) {
+            String singleRange = startEnd.single;
+            assert(singleRange.startsWith('U+'));
+            int rangeValue = int.parse(singleRange.substring(2), radix: 16);
+            fontFaceUnicodeRanges.add(_UnicodeRange(rangeValue, rangeValue));
+          } else {
+            assert(startEnd.length == 2);
+            String startRange = startEnd[0];
+            String endRange = startEnd[1];
+            assert(startRange.startsWith('U+'));
+            int startValue = int.parse(startRange.substring(2), radix: 16);
+            int endValue = int.parse(endRange, radix: 16);
+            fontFaceUnicodeRanges.add(_UnicodeRange(startValue, endValue));
+          }
+        }
+      } else if (line == '}') {
+        subsets.add(
+            _ResolvedNotoSubset(fontFaceUrl!, name, fontFaceUnicodeRanges!));
+        resolvingFontFace = false;
+      } else {
+        continue;
```

Comment:
Are we simply skipping or should we log parse warning at least to detect? Add a comment here.

Reply:
Done